securing_apps/topics/oidc/keycloak-gatekeeper.adoc

```diff
@@ -0,0 +1,494 @@
+
+==== Cross-origin resource sharing (CORS)
+
+You can add a CORS header via the `--cors-[method]` with these configuration options.
+
+ * Access-Control-Allow-Origin
+ * Access-Control-Allow-Methods
+ * Access-Control-Allow-Headers
+ * Access-Control-Expose-Headers
+ * Access-Control-Allow-Credentials
+ * Access-Control-Max-Age
+
+You can add using the config file:
+
+[source,yaml]
+----
+cors-origins:
+- '*'
+cors-methods:
+- GET
+- POST
+----
+
+or via the command line:
+
+[source,bash]
+----
+--cors-origins [--cors-origins option]                  a set of origins to add to the CORS access control (Access-Control-Allow-Origin)
+--cors-methods [--cors-methods option]                  the method permitted in the access control (Access-Control-Allow-Methods)
+--cors-headers [--cors-headers option]                  a set of headers to add to the CORS access control (Access-Control-Allow-Headers)
+--cors-exposes-headers [--cors-exposes-headers option]  set the expose cors headers access control (Access-Control-Expose-Headers)
+----
+
+==== Upstream URL
+
+You can control the upstream endpoint via the `--upstream-url` option. Both HTTP and HTTPS are supported with TLS verification and keep-alive support configured via the `--skip-upstream-tls-verify` / `--upstream-keepalives` option. Note, the proxy can also upstream via a UNIX socket, `--upstream-url unix://path/to/the/file.sock`.
+
+==== Endpoints
+
+* **/oauth/authorize** is authentication endpoint which will generate the OpenID redirect to the provider
+* **/oauth/callback** is provider OpenID callback endpoint
+* **/oauth/expired** is a helper endpoint to check if a access token has expired, 200 for ok and, 401 for no token and 401 for expired
+* **/oauth/health** is the health checking endpoint for the proxy, you can also grab version from headers
+* **/oauth/login** provides a relay endpoint to login via `grant_type=password`, for example, `POST /oauth/login` form values are `username=USERNAME&password=PASSWORD` (must be enabled)
+* **/oauth/logout** provides a convenient endpoint to log the user out, it will always attempt to perform a back channel log out of offline tokens
+* **/oauth/token** is a helper endpoint which will display the current access token for you
+* **/oauth/metrics** is a Prometheus metrics handler
+
+==== Metrics
+
+Assuming `--enable-metrics` has been set, a Prometheus endpoint can be found on */oauth/metrics*; at present the only metric being exposed is a counter per HTTP code.
+
+==== Limitations
+
+Keep in mind link:http://browsercookielimits.squawky.net/[browser cookie limits] if you use access or refresh tokens in the browser cookie. Keycloak-generic-adapter divides the cookie automatically if your cookie is longer than 4093 bytes. Real size of the cookie depends on the content of the issued access token. Also, encryption might add additional bytes to the cookie size. If you have large cookies (>200 KB), you might reach browser cookie limits.
```

Comment:
We need to replace `Keycloak-generic-adapter` by `Keycloak Gatekeeper`.

Reply:
Done.